prometheus/src/snapshot_metrics.rs

```diff
@@ -0,0 +1,65 @@
+use crate::utils::{write_metric, Metric, MetricFamily};
+use solana_runtime::snapshot_archive_info::{SnapshotArchiveInfo, SnapshotArchiveInfoGetter};
+use solana_runtime::snapshot_config::SnapshotConfig;
+use solana_runtime::snapshot_utils;
+use solana_sdk::clock::Slot;
+use std::io;
+
+pub fn write_snapshot_metrics<W: io::Write>(
+    snapshot_config: &Option<SnapshotConfig>,
+    out: &mut W,
+) -> io::Result<()> {
+    let snapshot_config = match snapshot_config.as_ref() {
+        Some(config) => config,
+        None => return Ok(()),
+    };
+
+    let full_snapshot_info = match get_full_snapshot_info(snapshot_config) {
+        Some(info) => info,
+        None => return Ok(()),
+    };
+    write_metric(
+        out,
+        &MetricFamily {
+            name: "solana_full_snapshot_info",
+            help: "The slot height of the most recent full snapshot",
+            type_: "counter",
```

Comment:
I guess there is no downside to using gauge as there might be some edge cases.

> The snapshot height might go down, for example if we delete the ledger directory and restart.

Well that would be only in case of some fork or network halt, no? I assume the snapshot would not be created until it catches up to the rest of the network. 

Did not notice we use gauge for slot hight tho, so it makes sense to use it here too.